tf2_ros/src/buffer.cpp

```diff
@@ -213,7 +213,7 @@ Buffer::waitForTransform(const std::string& target_frame, const std::string& sou
         std::lock_guard<std::mutex> lock(this->timer_to_request_map_mutex_);
         // Check if a timeout already occurred
         for (auto it = timer_to_request_map_.begin(); it != timer_to_request_map_.end(); ++it) {
-          if (request_handle == it->second) {
+          if (request_handle == it->second.request_handle) {
```

Comment:
This should be it->second.cb_handle
```suggestion
          if (request_handle == it->second.cb_handle) {
```

Reply:
request_handle is of type **tf2::TransformableRequestHandle**, cb_handle of **tf2::TransformableCallbackHandle**. I don't think we can compare them to each other.